monai/data/box_mode.py

```diff
@@ -0,0 +1,331 @@
+        spatial_dims = int(len(c) // 2)
+        return spatial_dims
+
+    def check_corner(self, c: Sequence) -> bool:
+        """
+        check the validity for the given box corners
+        Args:
+            c: corners of a box, 4-element or 6-element tuple, each element is a Nx1 torch tensor
+            (xmin, ymin, xmax, ymax) or (xmin, ymin, zmin, xmax, ymax, zmax)
+        Returns:
+            bool, whether the box is valid
+        Example:
+            c = (torch.ones(10,1), torch.ones(10,1), torch.ones(10,1), torch.ones(10,1))
+            boxmode.check_corner(c) will return True
+        """
+        spatial_dims = self.get_dim_from_corner(c)
+        box_error = c[spatial_dims] < c[0]
+        for axis in range(1, spatial_dims):
+            box_error = box_error | (c[spatial_dims + axis] < c[axis])
+        if box_error.sum() > 0:
+            return False
+        else:
+            return True
+
+    @abstractmethod
+    def box_to_corner(self, boxes: torch.Tensor) -> Tuple:
+        """
+        Return the box corners for the given boxes
+        Args:
+            boxes: bounding box, Nx4 or Nx6 torch tensor
+        Returns:
+            corners of a box, 4-element or 6-element tuple, each element is a Nx1 torch tensor
+        Example:
+            boxes = torch.ones(10,6)
+            boxmode.box_to_corner(boxes) will a 6-element tuple, each element is a 10x1 tensor
+        """
+        raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
+
+    @abstractmethod
+    def corner_to_box(self, corner: Sequence) -> torch.Tensor:
+        """
+        Return the boxes converted from the given box corners
+        Args:
+            c: corners of a box, 4-element or 6-element tuple, each element is a Nx1 torch tensor
+            (xmin, ymin, xmax, ymax) or (xmin, ymin, zmin, xmax, ymax, zmax)
+        Returns:
+            boxes: bounding box, Nx4 or Nx6 torch tensor
+        Example:
+            c = (torch.ones(10,1), torch.ones(10,1), torch.ones(10,1), torch.ones(10,1))
+            boxmode.corner_to_box(c) will return a 10x4 tensor
+        """
+        raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
+
+
+class CornerCornerModeTypeA(BoxMode):
+    """
+    Also represented as "xyxy" or "xyzxyz"
+    [xmin, ymin, xmax, ymax] or [xmin, ymin, zmin, xmax, ymax, zmax]
+    """
+
+    def __int__(self):
+        self.dim_to_str_mapping = {2: BoundingBoxMode.XYXY, 3: BoundingBoxMode.XYZXYZ}
+
+    def box_to_corner(self, boxes: torch.Tensor) -> Tuple:
+        spatial_dims = self.get_dim_from_boxes(boxes)
+        if spatial_dims == 3:
+            xmin, ymin, zmin, xmax, ymax, zmax = boxes.split(1, dim=-1)
+            corner = xmin, ymin, zmin, xmax, ymax, zmax
+        elif spatial_dims == 2:
+            xmin, ymin, xmax, ymax = boxes.split(1, dim=-1)
+            corner = xmin, ymin, xmax, ymax
+        if self.check_corner(corner):
```

Comment:
As the above comment, if put this `check_corner` function in the `box_utils`, I feel no need to check corner in every subclass of `BoxMode`.
How about just returning the converted corners (same as boxes, BTW seems you don't check valid box format?), and do the check in the outside logic?
What do you think?

Thanks.